packages/beacon-state-transition/src/allForks/util/epochProcess.ts

```diff
@@ -223,17 +223,23 @@ export function beforeProcessEpoch<T extends allForks.BeaconState>(state: Cached
       FLAG_CURR_HEAD_ATTESTER
     );
   } else {
-    state.previousEpochParticipation.forEachStatus((status, i) => {
-      statuses[i].flags |=
-        ((status.timelySource && FLAG_PREV_SOURCE_ATTESTER) as number) |
-        ((status.timelyTarget && FLAG_PREV_TARGET_ATTESTER) as number) |
-        ((status.timelyHead && FLAG_PREV_HEAD_ATTESTER) as number);
+    state.previousEpochParticipation.forEachStatus((participationStatus, i) => {
+      const status = statuses[i];
+      if (status.active) {
```

Comment:
i'm pretty sure this is referring to being `active` in the current epoch when you want to know if the validator was `active` in the **previous** epoch for this status pre-compute

Reply:
thanks Alex for having a look at this, I changed it accordingly. It's great to have someone from EF review this code! 🙏